WalletWasabi.Fluent/Models/WebcamQrReader.cs

```diff
@@ -0,0 +1,166 @@
+using Avalonia.Media.Imaging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using OpenCvSharp;
+using Avalonia;
+using Avalonia.Media;
+using System.Runtime.InteropServices;
+using WalletWasabi.Logging;
+using WalletWasabi.Userfacing;
+using NBitcoin;
+using System.Threading;
+
+namespace WalletWasabi.Fluent.Models
+{
+	public class WebcamQrReader
+	{
+		public bool RequestEnd { get; set; }
+		public Network Network { get; }
+		public Task? ScanningTask { get; set; }
+
+		public WebcamQrReader(Network network)
+		{
+			Network = network;
+		}
+
+		public void StartScanning()
+		{
+			if (ScanningTask is not { })
+			{
+				ScanningTask = Task.Run(() =>
+				{
+					VideoCapture? camera = null;
```

Comment:
Should this be disposed too as it implements `IDisposable`?

Reply:
`camera?.Release();` in the finally section should do the work.

Reply:
You're right, it's not really a dispose, but now I don't know if I should call both, or only `Dispose()`.
The documentation is not clear. Hopefully if I dispose an object, it would simply close the camera and let it go, right?

Reply:
https://csharpdoc.hotexamples.com/class/OpenCvSharp/VideoCapture#

It says `Release` closes video file or capturing device.
```
public void Release()
        {
            ThrowIfDisposed();
            NativeMethods.videoio_VideoCapture_release(ptr);
        }
```
I think we got to dispose it after releasing it.

Reply:
Yep, done that